src/paging.md

```diff
@@ -0,0 +1,534 @@
+Let’s get to it!
+
+```x86asm
+    ; point each page table level two entry to a page
+    mov ecx, 0         ; counter variable
+```
+
+In order to write a loop, we need a counter. `ecx` is the usual loop counter
+register, that’s what the `c` stands for: counter. We also have a comment
+indicating what we’re doing in this part of the code.
+
+Next, we need to make a new label:
+
+```x86asm
+.map_p2_table:
+```
+
+As we mentioned above, this is where we will loop back to when the loop
+continues.
+
+```x86asm
+    mov eax, 0x200000  ; 2MiB
+```
+
+We’re going to store 200,000 in `eax`. Here’s the reason: each page is two
+megabytes in size. So in order to get the right memory location, we will
+mutliply 200,000 by the number of the loop counter:
+
+
+|------------|---------|---------|---------|---------|---------|
+| counter    | 0       | 1       | 2       | 3       | 4       |
+| 200,000    | 200,000 | 200,000 | 200,000 | 200,000 | 200,000 |
+| multiplied | 0       | 200,000 | 400,000 | 600,000 | 800,000 |
+
+And so on. So our pages will be all next to each other, and 200,000 bytes in
+size.
+
+```x86asm
+    mul ecx
+```
+
+Here’s that multiplication! `mul` takes the value in `eax`, multiplies it by the
+register you’ve given it, and stores the result in `eax`. This will be the
+location of the next page.
+
+```x86asm
+    or eax, 0b10000011
+```
+
+Next up, our friend `or`. Here, we don’t just or `0b11`: we’re also setting
+another bit. This extra `1` is a ‘huge page’ bit, meaning that the pages are
+200,000 bytes. Without this bit, we’d have 4k pages instead of 2mb pages.
+
+```x86asm
+    mov [p2_table + ecx * 8], eax
+```
+
+Just like before, we are now writing the value in `eax` to a location. But
+instead of it being just `p2_table + 0`, we’re adding `ecx * 8` Remember, `ecx`
+is our loop counter. Each entry is eight bytes in size: `0b10000011`. So we need
+to multiply the counter by eight, and then add it to `p2_table`. Let’s take a
+closer look: let’s assume `p2_table` is zero, to make the math easier:
+
+|---------------------|---------|---------|---------|---------|---------|
+| p2\_table           | 0       | 0       | 0       | 0       | 0       |
+| ecx                 | 0       | 1       | 2       | 3       | 4       |
+| ecx * 8             | 0       | 8       | 16      | 24      | 32      |
+| p2\_table + ecx * 8 | 0       | 8       | 16      | 24      | 32      |
+
+We skip eight spaces each time, so we have room for all eight bits of the page
+table.
+
+That’s the body of the loop! Now we need to see if we need to keep looping or
+not:
+
+```x86asm
+    inc ecx
+    cmp ecx, 512
+    jne .map_p2_table
+```
+
+The `inc` statement increments the register it’s given by one. `ecx` is our loop
+counter, so we’re adding to it. Then, we ‘compare’ with `cmp`. We’re comparing
+`ecx` with 512: we want to map 512 page entries overall. This will give us 512 *
+2 megabytes: one gigabyte of memory. It’s also why we wrote the loop: writing
```

Comment:
(Here and elsewhere)
Should you use "mebibyte" and "gibibyte" instead of "mega", "giga"?